ams/ams-server/src/main/java/com/netease/arctic/ams/server/controller/CatalogController.java

```diff
@@ -0,0 +1,427 @@
+ * limitations under the License.
+ */
+
+package com.netease.arctic.ams.server.controller;
+
+import com.netease.arctic.ams.api.CatalogMeta;
+import com.netease.arctic.ams.api.properties.CatalogMetaProperties;
+import com.netease.arctic.ams.api.properties.TableFormat;
+import com.netease.arctic.ams.server.config.ConfigFileProperties;
+import com.netease.arctic.ams.server.controller.response.ErrorResponse;
+import com.netease.arctic.ams.server.controller.response.OkResponse;
+import com.netease.arctic.ams.server.model.CatalogRegisterInfo;
+import com.netease.arctic.ams.server.model.CatalogSettingInfo;
+import com.netease.arctic.ams.server.model.CatalogSettingInfo.ConfigFileItem;
+import com.netease.arctic.ams.server.service.IMetaService;
+import com.netease.arctic.ams.server.service.ServiceContainer;
+import com.netease.arctic.ams.server.service.impl.CatalogMetadataService;
+import com.netease.arctic.ams.server.service.impl.PlatformFileInfoService;
+import io.javalin.http.Context;
+import io.javalin.http.HttpCode;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.netease.arctic.ams.api.properties.CatalogMetaProperties.CATALOG_TYPE_CUSTOM;
+
+public class CatalogController extends RestBaseController {
+  private static final Logger LOG = LoggerFactory.getLogger(CatalogController.class);
+  private static final IMetaService iMetaService = ServiceContainer.getMetaService();
+  private static CatalogMetadataService catalogMetadataService = ServiceContainer.getCatalogMetadataService();
+  private static PlatformFileInfoService platformFileInfoService = ServiceContainer.getPlatformFileInfoService();
+
+  private static final String AUTH_CONFIG_KEY_TYPE = "auth_config.type";
+  private static final String AUTH_CONFIG_TYPE_VALUE_SIMPLE = "simple";
+  private static final String AUTH_CONFIG_TYPE_VALUE_KERBEROS = "kerberos";
+  private static final String AUTH_CONFIG_KEY_KEYTAB = "auth_config.keytab";
+  private static final String AUTH_CONFIG_KEY_PRINCIPAL = "auth_config.principal";
+  private static final String AUTH_CONFIG_HADOOP_USERNAME = "auth_config.hadoop_username";
+  private static final String AUTH_CONFIG_KEY_KRB5 = "auth_config.krb5";
+
+  private static final String CONFIG_TYPE_STORAGE = "storage-config";
+  private static final String CONFIG_TYPE_AUTH = "auth-config";
+
+  /**
+   * get catalog Type list
+   *
+   * @param ctx
+   */
+  public static void getCatalogTypeList(Context ctx) {
+
+    List<Map<String, String>> catalogTypeList = new ArrayList<>();
+    String valueKey = "value";
+    String displayKey = "display";
+    List<List<String>> valuePair = new ArrayList<List<String>>() {
+      {
+        add(Arrays.asList(CatalogMetaProperties.CATALOG_TYPE_HIVE, "Hive Metastore"));
```

Comment:
You may simplify your code with `valuePair.add(ImmutableMap.of(valueKey, CatalogMetaProperties.CATALOG_TYPE_HIVE, displayKey, "Hive Metastore"))`

Reply:
> You may simplify your code with `valuePair.add(ImmutableMap.of(valueKey, CatalogMetaProperties.CATALOG_TYPE_HIVE, displayKey, "Hive Metastore"))`

done